tests/test_ndarray.py

```diff
@@ -92,3 +92,33 @@ def test_Gatherv_random_rows():
 
     if rank == root:
         assert(data.shape[0] == np.sum(sizes))
+
+
+@pytest.mark.skipif(MPI is None, reason='MPI not installed.')
+def test_Gather_ndlist():
+    """Test Gather_ndlist for concatenating lists of various lengths,
+    with each element being an nd-array of various dimension"""
+
+    comm = MPI.COMM_WORLD
+    root = 0
+    rank = comm.rank
+    numprocs = comm.size
+
+    # generate list of differently sized np arrays
+    size = rank + 1
+    data = [rank * np.ones((size, size)) for i in range(size)]
```

Comment:
I think you should something more like `np.arange(size**2).reshape(size, size)` so that you can check that both the shape and values are correct. Raveling and reshaping could potentially swap axes.